includes/init_includes/init_languages.php

```diff
@@ -20,6 +20,7 @@
   } else {
     if (LANGUAGE_DEFAULT_SELECTOR=='Browser') {
       $lng->get_browser_language();
+      if (!zen_not_null($lng->language['id'])) $lng->set_language(DEFAULT_LANGUAGE);

```

Comment:
Had considered as well, but thought the "purity" of the function should be preserved (ie. what information is actually provided from the browser) and based on the response of that "object" to take the applicable action although some of what is "taken" is directly related to what is in the store's database, so certainly on the line with that "tough question". :) 

There are even alternative directions that could be taken, as the "default" code that follows basically assumes a vanilla install (english, 1, en) where a similar "search" could be performed for "those" specific values (expecting english to remain unchanged) and if not found then fallback on to the default language (assuming that its definition hasn't been lost.)

Sorry, I had a number of potential directions to consider as a solution, but determined that all other paths seemed to lead to the DEFAULT_LANGUAGE and there exists code to handle it therefore if the specific language_id isn't assigned then tried to be sure a language_id was identified without entering an infinite loop.

Do see that the admin side somewhat already forces DEFAULT_LANGUAGE even for browser detection.